### PR TITLE
fix: Sanitize Wearables and Scene files

### LINKED ISSUE #61: [BUG] Detecting SW files
When loading a .zip SW file that contains a few folders inside, e.g.:
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBa1kzIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--057c5d84fa6129534e4b0e171e91367261c602ba/image.png)

The handleZippedModelFiles method doesn't detect the `scene.json` and `wearable.json` files correctly, so it returns the wearable without the scene.json references.

Tasks:
- [ ] Update the loading files process to sanitize the folders and be `scene.json` and `wearable.json` at the root level

### PR DESCRIPTION
**Pull request by cyaiox**
This PR sanitizes the files from a .zip file containing parent folders.

Closes #61

### CODE REVIEW

File: src/files/files.ts [hunk 1]
```diff
@@ -105,20 +105,30 @@ async function handleZippedModelFiles<T extends Content>(
...
+    ) {
+      baseDir = dirname(filePath) === '.' ? '' : `${dirname(filePath)}/`
     }
   })
 
   const fileContents = await Promise.all(promiseOfFileContents)
   const content: RawContent<T> = fileNames.reduce((acc, fileName, index) => {
+    if (fileName.endsWith('/')) {
+      return acc
+    }
```
Comment (LautaroPetaccio) [inline src/files/files.ts:130]: When would a filename end with `/`?
Is the library listing directories?

File: src/files/files.ts [hunk 2]
```diff
@@ -105,20 +105,30 @@ async function handleZippedModelFiles<T extends Content>(
...
-      basename(filePath) !== BUILDER_MANIFEST &&
-      basename(filePath) !== SCENE_MANIFEST
+      ![WEARABLE_MANIFEST, BUILDER_MANIFEST, SCENE_MANIFEST].includes(
+        basename(filePath)
+      )
     ) {
       fileNames.push(filePath)
       promiseOfFileContents.push(file.async(fileFormat) as Promise<T>)
+    } else if (
+      [WEARABLE_MANIFEST, SCENE_MANIFEST].includes(basename(filePath))
```
Comment (LautaroPetaccio) [inline src/files/files.ts:120]: Could it happen for these two files to be in different directories?

Comment (cyaiox) [reply src/files/files.ts:120]: it shouldn't happen. both files have to be at the same level.

I will add an extra validation to ensure both files are in the same directory

File: src/files/files.ts (hunk 1 above)
Comment (cyaiox) [reply src/files/files.ts:130]: > Is the library listing directories?
yeap, it's the library listing direcotries

File: src/files/files.ts [hunk 3]
```diff
@@ -131,16 +141,18 @@ async function handleZippedModelFiles<T extends Content>(
...
   }, {})
 
   let wearable: WearableConfig | undefined = undefined
   let scene: SceneConfig | undefined = undefined
   let builder: BuilderConfig | undefined = undefined
 
-  const wearableZipFile = zip.file(WEARABLE_MANIFEST)
-  const sceneZipFile = zip.file(SCENE_MANIFEST)
+  const wearableZipFile = zip.file(new RegExp(`${WEARABLE_MANIFEST}$`))[0]
+  const sceneZipFile = zip.file(new RegExp(`${SCENE_MANIFEST}$`))[0]
```
[the remark this answers is not in the document]
Comment (cyaiox) [reply src/files/files.ts:155]: I like it, will try it

**Comment by cyaiox** [comment]
Fixed here: https://github.com/decentraland/js-sdk-toolchain/pull/714

**Pull request closed without merging**